### ISSUE TITLE
Crashes when importing specific files because of cuesheets

### ISSUE THREAD

**Issue by 4oo4** [DeaDBeeF-Player/deadbeef#2493]
### Steps to reproduce the problem
Load specific files into library, deadbeef crashes. Out of thousands of flac files, only a few hundred of them seem to cause this, and after testing them with a debug build they all crash in the exact same way.

### What's going on? Describe the problem in as much detail as possible.
It seems to be something with a cuesheet that deadbeef doesn't like. When I remove a .cue file from one of the folders I'm trying to import it works normally.

Part of the trace mentions an issue with an embedded cuesheet, though when I look at one of the files with `metaflac` I don't think there's a cuesheet block, just the separate .cue file:

https://github.com/DeaDBeeF-Player/deadbeef/blob/7c04f04e64350e3d63e2776e326d82d42f92d79d/cueutil.c#L834-L837

**Flac tag example**:
```
METADATA block #0
  type: 0 (STREAMINFO)
  is last: false
  length: 34
  minimum blocksize: 4096 samples
  maximum blocksize: 4096 samples
  minimum framesize: 3950 bytes
  maximum framesize: 7677 bytes
  sample_rate: 96000 Hz
  channels: 1
  bits-per-sample: 24
  total samples: 30153600
  MD5 signature: 73f6b79da20cbf68ff3c2aae2e217d66
METADATA block #1
  type: 3 (SEEKTABLE)
  is last: false
  length: 576
  seek points: 32
    point 0: sample_number=0, stream_offset=0, frame_samples=4096
    --snip--
    point 31: sample_number=29757440, stream_offset=46448461, frame_samples=4096
METADATA block #2
  type: 4 (VORBIS_COMMENT)
  is last: false
  length: 335
  vendor string: reference libFLAC 1.3.2 20170101
  comments: 13
    comment[0]: artist=Stan Getz, João Gilberto
    comment[1]: album=Getz/Gilberto
    comment[2]: genre=Jazz
    comment[3]: date=1964
    comment[4]: title=The Girl From Ipanema
    comment[5]: albumartist=Stan Getz, João Gilberto
    comment[6]: comment=Verve Records / V-8545 / LP / Mono
    comment[7]: tracknumber=1
    comment[8]: totaltracks=8
    comment[9]: tracktotal=8
    comment[10]: disc=1
    comment[11]: discnumber=1
    comment[12]: disctotal=1
METADATA block #3
  type: 1 (PADDING)
  is last: true
  length: 8192
```

Looking at the cuesheet, I don't see anything wrong with it, unless I'm missing something.

**Cuesheet**
```
PERFORMER "Stan Getz"
TITLE "Stan Getz  João Gilberto Featuring Antonio Carlos Jobim"
REM StigLarsson MONO
FILE "Stan Getz  João Gilberto.flac" flac
  TRACK 01 AUDIO
    TITLE "The Girl From Ipanema"
    INDEX 01 00:00:00
  TRACK 02 AUDIO
    TITLE "Doralice"
    INDEX 01 05:14:10
  TRACK 03 AUDIO
    TITLE "Para Machuchar Meu Coração"
    INDEX 01 07:59:13
  TRACK 04 AUDIO
    TITLE "Desafinado"
    INDEX 01 13:06:12
  TRACK 05 AUDIO
    TITLE "Corcovado"
    INDEX 01 17:17:06
  TRACK 06 AUDIO
    TITLE "Só Danço Samba"
    INDEX 01 21:33:01
  TRACK 07 AUDIO
    TITLE "O Grande Amor"
    INDEX 01 24:56:07
  TRACK 08 AUDIO
    TITLE "Vivo Sonhando"
    INDEX 01 30:22:11
```

### Information about the software:

Deadbeef version: 1.8.4
OS: Debian 10

### Stack Trace:

```
GNU gdb (Debian 8.2.1-2+b3) 8.2.1
Copyright (C) 2018 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.
Type "show copying" and "show warranty" for details.
This GDB was configured as "x86_64-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
    <http://www.gnu.org/software/gdb/documentation/>.

For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from deadbeef...done.
(gdb) r
Starting program: /home/user/src/deadbeef/deadbeef 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
starting deadbeef 1.8.4
server_start
searching for GUI plugins in /home/user/.local/lib64/deadbeef
searching for GUI plugins in /home/user/.local/lib/deadbeef
load_plugin_dir /home/user/.local/lib/deadbeef: scandir found 6 files
searching for GUI plugins in /usr/local/lib/deadbeef
load_plugin_dir /usr/local/lib/deadbeef: scandir found 76 files
found gui plugin ddb_gui_GTK2.so
added GTK2 gui plugin
found gui plugin ddb_gui_GTK3.so
added GTK3 gui plugin
load gui plugin
checking GUI plugin: GTK2
checking GUI plugin: GTK3
found selected GUI plugin: GTK3
loading plugin /usr/local/lib/deadbeef/ddb_gui_GTK3.so
loading plugins from /home/user/.local/lib64/deadbeef
loading plugins from /home/user/.local/lib/deadbeef
load_plugin_dir /home/user/.local/lib/deadbeef: scandir found 6 files
loading plugin /home/user/.local/lib/deadbeef/ddb_misc_waveform_GTK2.so
loading plugin /home/user/.local/lib/deadbeef/ddb_misc_waveform_GTK3.so
loading plugin /home/user/.local/lib/deadbeef/ddb_vis_spectrogram_GTK2.so
loading plugin /home/user/.local/lib/deadbeef/ddb_vis_spectrogram_GTK3.so
loading plugins from /usr/local/lib/deadbeef
load_plugin_dir /usr/local/lib/deadbeef: scandir found 76 files
loading plugin /usr/local/lib/deadbeef/adplug.so
loading plugin /usr/local/lib/deadbeef/alac.so
loading plugin /usr/local/lib/deadbeef/converter.so
loading plugin /usr/local/lib/deadbeef/converter_gtk2.so
loading plugin /usr/local/lib/deadbeef/converter_gtk3.so
loading plugin /usr/local/lib/deadbeef/dca.so
loading plugin /usr/local/lib/deadbeef/ddb_dumb.so
loading plugin /usr/local/lib/deadbeef/ddb_mono2stereo.so
loading plugin /usr/local/lib/deadbeef/ddb_shn.so
loading plugin /usr/local/lib/deadbeef/ddb_soundtouch.so
loading plugin /usr/local/lib/deadbeef/gme.so
loading plugin /usr/local/lib/deadbeef/hotkeys.so
loading plugin /usr/local/lib/deadbeef/in_sc68.so
loading plugin /usr/local/lib/deadbeef/lastfm.so
loading plugin /usr/local/lib/deadbeef/m3u.so
loading plugin /usr/local/lib/deadbeef/mms.so
loading plugin /usr/local/lib/deadbeef/musepack.so
loading plugin /usr/local/lib/deadbeef/notify.so
loading plugin /usr/local/lib/deadbeef/nullout.so
loading plugin /usr/local/lib/deadbeef/oss.so
loading plugin /usr/local/lib/deadbeef/pltbrowser_gtk2.so
loading plugin /usr/local/lib/deadbeef/pltbrowser_gtk3.so
loading plugin /usr/local/lib/deadbeef/psf.so
loading plugin /usr/local/lib/deadbeef/rg_scanner.so
loading plugin /usr/local/lib/deadbeef/shellexec.so
loading plugin /usr/local/lib/deadbeef/shellexecui_gtk2.so
loading plugin /usr/local/lib/deadbeef/shellexecui_gtk3.so
loading plugin /usr/local/lib/deadbeef/sid.so
loading plugin /usr/local/lib/deadbeef/supereq.so
loading plugin /usr/local/lib/deadbeef/tta.so
loading plugin /usr/local/lib/deadbeef/vfs_curl.so
loading plugin /usr/local/lib/deadbeef/vtx.so
loading plugin /usr/local/lib/deadbeef/wildmidi.so
loading plugin /usr/local/lib/deadbeef/wma.so
starting plugin GTK3 user interface
starting plugin Waveform Seekbar
starting plugin Waveform Seekbar
starting plugin Spectrogram
starting plugin Spectrogram
starting plugin Adplug player
starting plugin ALAC player
starting plugin Converter
starting plugin Converter GTK2 UI
starting plugin Converter GTK3 UI
starting plugin dts decoder
starting plugin DUMB module player
starting plugin Mono to stereo
starting plugin Shorten player
starting plugin Soundtouch
starting plugin Game-Music-Emu player
starting plugin Hotkey manager
[New Thread 0x7ffff26cd700 (LWP 23698)]
starting plugin SC68 player (Atari ST SNDH YM2149)
starting plugin last.fm scrobbler
[New Thread 0x7ffff1ecc700 (LWP 23699)]
starting plugin M3U and PLS support
starting plugin mms vfs
starting plugin MusePack decoder
starting plugin OSD Notify
starting plugin Null output plugin
starting plugin OSS output plugin
starting plugin Playlist browser GTK2
starting plugin Playlist browser GTK3
starting plugin PSF player using Audio Overload SDK
starting plugin ReplayGain Scanner
starting plugin Shell commands
starting plugin Shellexec GTK2 UI
starting plugin Shellexec GTK3 UI
starting plugin SID player
starting plugin SuperEQ
starting plugin tta decoder
starting plugin cURL vfs
starting plugin VTX player
starting plugin WildMidi player
starting plugin WMA player
starting plugin stdio vfs
selected output plugin: Null output plugin
INFO: from file /home/user/.config/deadbeef/playlists/0.dbpl
INFO: from file /home/user/.config/deadbeef/playlists/1.dbpl
streamer_dsp_chain_load: plugin SRC not found. preset will not be loaded
streamer_dsp_chain_load: error loading /home/user/.config/deadbeef/dspconfig
[New Thread 0x7ffff168a700 (LWP 23700)]
plugin Waveform Seekbar failed to connect to dependencies, deactivated.
plugin Spectrogram failed to connect to dependencies, deactivated.
convgui: gtkui plugin not found
plugin Converter GTK2 UI failed to connect to dependencies, deactivated.
plugin Playlist browser GTK2 failed to connect to dependencies, deactivated.
shellexecui: can't find gtkui plugin
plugin Shellexec GTK2 UI failed to connect to dependencies, deactivated.
resume: track -1 pos -1.000000 playlist -1
[New Thread 0x7ffff0e89700 (LWP 23701)]
[New Thread 0x7ffff0688700 (LWP 23702)]
gtkui plugin compiled for gtk version: 3.24.5
[New Thread 0x7fffefe87700 (LWP 23703)]
[New Thread 0x7fffef686700 (LWP 23704)]
connecting button tray signals
[New Thread 0x7fffeee05700 (LWP 23705)]

(deadbeef:23694): GLib-CRITICAL **: 19:52:35.970: Source ID 25 was not found when attempting to remove it
[Thread 0x7fffeee05700 (LWP 23705) exited]
[New Thread 0x7fffeee05700 (LWP 23706)]
[New Thread 0x7fffedc6b700 (LWP 23707)]

Thread 10 "deadbeef-main" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffeee05700 (LWP 23706)]
0x0000000000418f9c in pl_item_get_startsample (it=0x0) at playlist.c:3890
3890	    if (!it->has_startsample64) {
(gdb) Quit
(gdb) thread apply all backtrace

Thread 11 (Thread 0x7fffedc6b700 (LWP 23707)):
#0  0x00007ffff7d32f59 in syscall () at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
#1  0x00007ffff6a100ba in g_cond_wait_until () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff6998091 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff69eeef2 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff69ee415 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#6  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 10 (Thread 0x7fffeee05700 (LWP 23706)):
#0  0x0000000000418f9c in pl_item_get_startsample (it=0x0) at playlist.c:3890
#1  0x000000000040b4a3 in plt_load_cuesheet_from_buffer
    (plt=0x5323e0, after=0x8559280, fname=0x7fffeee01ad0 "/media/share/Music/Stan Getz & João Gilberto - 1964 - Getz-Gilberto/Stan Getz, João Gilberto - Getz-Gilberto.cue", embedded_origin=0x0, embedded_numsamples=0, embedded_samplerate=0, buffer=0x7fffd8006800 "PERFORMER \"Stan Getz\"\r\nTITLE \"Stan Getz  Jo\343o Gilberto Featuring Antonio Carlos Jobim\"\r\nREM StigLarsson MONO\r\nFILE \"Stan Getz  Jo\343o Gilberto.flac\" flac\r\n  TRACK 01 AUDIO\r\n    TITLE \"The Girl From Ipan"..., sz=709, dirname=0x7fffeee02bd0 "/media/share/Music/Stan Getz & João Gilberto - 1964 - Getz-Gilberto", namelist=0x7fffd8001800, n=14) at cueutil.c:835
#2  0x000000000040af01 in plt_load_cue_file
    (plt=0x5323e0, after=0x8559280, fname=0x7fffeee01ad0 "/media/share/Music/Stan Getz & João Gilberto - 1964 - Getz-Gilberto/Stan Getz, João Gilberto - Getz-Gilberto.cue", dirname=0x7fffeee02bd0 "/media/share/Music/Stan Getz & João Gilberto - 1964 - Getz-Gilberto", namelist=0x7fffd8001800, n=14) at cueutil.c:483
#3  0x00000000004114fd in plt_insert_dir_int
    (visibility=0, playlist=0x5323e0, vfs=0x0, after=0x8559280, dirname=0x7fffeee04d87 "/media/share/--Type <RET> for more, q to quit, c to continue without paging--c
Music/Stan Getz & João Gilberto - 1964 - Getz-Gilberto", pabort=0x7fffeee04e7c, cb=0x0, user_data=0x0) at playlist.c:1251
#4  0x0000000000418c7f in plt_insert_dir2 (visibility=0, plt=0x5323e0, after=0x8559280, dirname=0x7fffeee04d80 "file:///media/share/Music/Stan Getz & João Gilberto - 1964 - Getz-Gilberto", pabort=0x7fffeee04e7c, callback=0x0, user_data=0x0) at playlist.c:3775
#5  0x00007ffff77229f4 in gtkpl_add_fm_dropped_files (drop_before=0x0, ptr=0x8af4c50 "file:///media/share/Music/Stan%20Getz%20&%20Jo%C3%A3o%20Gilberto%20-%201964%20-%20Getz-Gilberto%20%5BFLAC-24-96,%20V-8545,%20LP,%20Mono%5D\r\n", length=159) at fileman.c:235
#6  0x00007ffff7722e3f in fmdrop_worker (ctx=0x8c99170) at fileman.c:279
#7  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#8  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 8 (Thread 0x7fffef686700 (LWP 23704)):
#0  0x00007ffff7d2d819 in __GI___poll (fds=0x4ab740, nfds=3, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff69c6136 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff69c64c2 in g_main_loop_run () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff6bf2266 in  () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00007ffff69ee415 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#6  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 7 (Thread 0x7fffefe87700 (LWP 23703)):
#0  0x00007ffff7d2d819 in __GI___poll (fds=0x50d2a0, nfds=2, timeout=-1) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff69c6136 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff69c625c in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff69c62a1 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#4  0x00007ffff69ee415 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#5  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#6  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 6 (Thread 0x7ffff0688700 (LWP 23702)):
#0  0x00007ffff7e0e00c in futex_wait_cancelable (private=0, expected=0, futex_word=0x4a7dbc) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  0x00007ffff7e0e00c in __pthread_cond_wait_common (abstime=0x0, mutex=0x4a7d60, cond=0x4a7d90) at pthread_cond_wait.c:502
#2  0x00007ffff7e0e00c in __pthread_cond_wait (cond=0x4a7d90, mutex=0x4a7d60) at pthread_cond_wait.c:655
#3  0x0000000000426e4f in cond_wait (c=4881808, m=4881760) at threading_pthread.c:221
#4  0x000000000042531d in messagepump_wait () at messagepump.c:130
#5  0x0000000000405535 in player_mainloop () at main.c:829
#6  0x0000000000407521 in mainloop_thread (ctx=0x0) at main.c:963
#7  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#8  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 5 (Thread 0x7ffff0e89700 (LWP 23701)):
#0  0x00007ffff7d30037 in __GI___select (nfds=4, readfds=0x7ffff0e88e68, writefds=0x0, exceptfds=0x0, timeout=0x7ffff0e88e50) at ../sysdeps/unix/sysv/linux/select.c:41
#1  0x00000000004050fe in server_loop (ctx=0x0) at main.c:692
#2  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#3  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 4 (Thread 0x7ffff168a700 (LWP 23700)):
#0  0x00007ffff7d05720 in __GI___nanosleep (requested_time=requested_time@entry=0x7ffff1689e10, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
#1  0x00007ffff7d30874 in usleep (useconds=<optimized out>) at ../sysdeps/posix/usleep.c:32
#2  0x000000000041b998 in streamer_thread (unused=0x0) at streamer.c:1588
#3  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#4  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 3 (Thread 0x7ffff1ecc700 (LWP 23699)):
#0  0x00007ffff7e0e00c in futex_wait_cancelable (private=0, expected=0, futex_word=0x52cb48) at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  0x00007ffff7e0e00c in __pthread_cond_wait_common (abstime=0x0, mutex=0x4daf60, cond=0x52cb20) at pthread_cond_wait.c:502
#2  0x00007ffff7e0e00c in __pthread_cond_wait (cond=0x52cb20, mutex=0x4daf60) at pthread_cond_wait.c:655
#3  0x0000000000426e4f in cond_wait (c=5425952, m=5091168) at threading_pthread.c:221
#4  0x00007ffff49975a7 in lfm_thread (ctx=0x0) at lastfm.c:744
#5  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#6  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 2 (Thread 0x7ffff26cd700 (LWP 23698)):
#0  0x00007ffff7d05720 in __GI___nanosleep (requested_time=requested_time@entry=0x7ffff26ccdc0, remaining=remaining@entry=0x0) at ../sysdeps/unix/sysv/linux/nanosleep.c:28
#1  0x00007ffff7d30874 in usleep (useconds=<optimized out>) at ../sysdeps/posix/usleep.c:32
#2  0x00007ffff3b2fcfb in hotkeys_event_loop (unused=0x0) at hotkeys.c:501
#3  0x00007ffff7e07fa3 in start_thread (arg=<optimized out>) at pthread_create.c:486
#4  0x00007ffff7d384cf in clone () at ../sysdeps/unix/sysv/linux/x86_64/clone.S:95

Thread 1 (Thread 0x7ffff7c3db80 (LWP 23694)):
#0  0x00007ffff7d2d819 in __GI___poll (fds=0x889dfe0, nfds=3, timeout=25) at ../sysdeps/unix/sysv/linux/poll.c:29
#1  0x00007ffff69c6136 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#2  0x00007ffff69c625c in g_main_context_iteration () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#3  0x00007ffff6bb8a2d in g_application_run () at /lib/x86_64-linux-gnu/libgio-2.0.so.0
#4  0x00007ffff76e87e7 in gtkui_start () at gtkui.c:1515
#5  0x0000000000406e9f in main (argc=1, argv=0x7fffffffdd18) at main.c:1348
```

**Comment by Alexey-Yakovenko** [comment]
Would you be so kind to check if the issue is still valid in 1.8.5-rc1?
https://sourceforge.net/projects/deadbeef/files/travis/linux/1.8/deadbeef-static_1.8.5-rc1-1_x86_64.tar.bz2/download

**Comment by 4oo4** [comment]
Yep, I just tested that build and I can no longer reproduce it. Just curious what the fix was?

Cheers

**Comment by Alexey-Yakovenko** [comment]
Most likely was fixed as part of issue #2389
